realm/src/main/java/io/realm/Realm.java

```diff
@@ -1477,6 +1487,7 @@ public static void migrateRealm(RealmConfiguration configuration, RealmMigration
             realm.beginTransaction();
             realm.setVersion(realmMigration.execute(realm, realm.getVersion()));
             realm.commitTransaction();
+            validatedRealmFiles.remove(configuration.getPath());
```

Comment:
While there is a race condition in close that could cause this to be called twice, that doesn't really matter as calling remove on something that isn't there would be a no-op. Also the backing set is Thread safe, so no there shouldn't be any need for a write transaction.